tests/unit/test_config.py

```diff
@@ -232,7 +243,39 @@ def test_legacy_variables_override_if_given(self, default_ip):
             edge_port_http,
         ) = config.populate_legacy_edge_configuration(environment)
 
-        assert gateway_listen == f"{default_ip}:4566"
+        assert gateway_listen == [HostAndPort(host=default_ip, port=4566)]
         assert edge_bind_host == "192.168.0.1"
         assert edge_port == 10101
         assert edge_port_http == 20202
+
+    # generate random hostnames and ports from within the following criteria:
+    # - hostnames: a-zA-Z0-9-.
+    # - ports: 0-65535
+    @given(text(alphabet=SAMPLER, min_size=1), integers(min_value=0, max_value=2**16 - 1))
```

Comment:
i'm generally not in favor of adding a library for one test TBH, especially if there's no clear outlook of using it elsewhere. IMO this is such basic functionality and i wonder whether it's really necessary to add this complexity. i'm generally a fan of test cases being obvious, and especially when testing parsing methods i think it's better the asserts just sit there when i look at the test.

i'm not going to be contentious on the matter, if @simonrw you feel it's a good addition, then let's leave it.